src/Symfony/Component/Lock/Lock.php

```diff
@@ -36,13 +36,17 @@ final class Lock implements LockInterface, LoggerAwareInterface
     private $dirty = false;
 
     /**
-     * @param Key            $key         Resource to lock
-     * @param StoreInterface $store       Store used to handle lock persistence
-     * @param float|null     $ttl         Maximum expected lock duration in seconds
-     * @param bool           $autoRelease Whether to automatically release the lock or not when the lock instance is destroyed
+     * @param Key                                                                  $key         Resource to lock
+     * @param PersistStoreInterface|BlockingStoreInterface|ExpirableStoreInterface $store       Store used to handle lock persistence
```

Comment:
IMHO it's not a "or" `|`, it should be a `PersistStoreInterface` + optional interfaces